### PR TITLE
Modified ssdhealth test to account for currently supported disk types

### PR DESCRIPTION
**Pull request by assrinivasan**
### Description of PR

Summary:
This is a sonic-mgmt test for [sonic-buildimage issue 9407](https://github.com/sonic-net/sonic-buildimage/issues/9407), which has a fix PR raised here: https://github.com/sonic-net/sonic-utilities/pull/3399 -- the change in this PR tests that fix.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [x] 202012
- [x] 202205
- [x] 202305
- [x] 202311
- [x] 202405

#### What is the motivation for this PR?
`show platform ssdhealth` fails on certain platforms such as Arista 7050qx because the sonic utility that the command calls relies on `smartctl` output, which fails on storage device of type EUSB. This PR skips the test on unsupported disk types.

#### How did you do it?
Added a new field `Disk Type` to the `ssdutil` sonic-utility that the `show platform ssdhealth` calls. The test checks for a supported disk type and skips if not found.

#### How did you verify/test it?
Tested on 4 HWSKUs each with storage disk type EUSB (skip), EMMC (skip), SATA (pass) and NVME (pass). Logs are attached here: [test_show_platform_ssdhealth.txt](https://github.com/user-attachments/files/16571432/test_show_platform_ssdhealth.txt)

#### Any platform specific information?
Skipped on platforms with storage device type EUSB and EMMC.

### CODE REVIEW

**Comment by assrinivasan** [comment]
Test failure appears to not be connected with this PR.

File: tests/platform_tests/cli/test_show_platform.py [hunk 1]
```diff
@@ -447,11 +447,15 @@ def test_show_platform_ssdhealth(duthosts, enum_supervisor_dut_hostname):
     """
     duthost = duthosts[enum_supervisor_dut_hostname]
     cmd = " ".join([CMD_SHOW_PLATFORM, "ssdhealth"])
+    supported_disks = ["SATA", "NVME"]
 
     logging.info("Verifying output of '{}' on ''{}'...".format(cmd, duthost.hostname))
+
     ssdhealth_output_lines = duthost.command(cmd)["stdout_lines"]
+    if not any(disk_type in ssdhealth_output_lines[0] for disk_type in supported_disks):
+        pytest.skip("Disk Type is not supported")
```
Comment (prgeor) [inline tests/platform_tests/cli/test_show_platform.py:456]: @assrinivasan print what is the platform's disk type ?

Comment (assrinivasan) [reply tests/platform_tests/cli/test_show_platform.py]: Changed in latest.

**Review by prgeor — APPROVED** [review]

**Comment by weiguo-nvidia** [comment]
@prgeor , may I ask is there any plan to merge the fixe to 202405 branch?

**Comment by yejianquan** [comment]
After https://github.com/sonic-net/sonic-utilities/pull/3399 got cherry-picke and advanced into 202405 branch, platform_tests/cli/test_show_platform.py reports failure due to mismatch

**Pull request merged**